### PR TITLE
fix: Set BitmapSource's PixelWidth/PixelHeight properly

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs [hunk 1]
```diff
@@ -16,16 +16,17 @@ public partial class BitmapSource : ImageSource
 		public int PixelHeight
 		{
 			get { return (int)GetValue(PixelHeightProperty); }
-			internal set { SetValue(PixelHeightProperty, value); }
+#if __WASM__ // The setter should be private protected and must not be set by Image. In fact, the PixelWidth/PixelHeight value should be available even without being rendered by Image.
+			internal
+#else
+			private protected
+#endif
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs:23]: Note to self that I don't forget:

This comment is incorrect. `Image` should update PixelWidth/PixelHeight when BitmapImage is constructed via a Uri.

This should be done before raising BitmapImage.ImageOpened here:

https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Controls/Image/Image.cs#L100

https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.skia.cs#L92


https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs#L189

**Review by Youssef1313 — COMMENTED** [review]
TODO (note to self):

Use naturalWidth/naturalHeight for Wasm: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/naturalWidth

**Comment by Youssef1313** [comment]
TODO: Make sure this fixes https://github.com/unoplatform/uno/issues/5238

File: src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs [hunk 2]
```diff
@@ -96,19 +87,20 @@ private void SetSourceCore(IRandomAccessStream streamSource)
...
 #if __NETSTD__
 			_stream = clonedStreamSource;
+			UpdatePixelWidthAndHeightPartial(_stream.AsStream());
 #else
 			Stream = clonedStreamSource.AsStream();
+			UpdatePixelWidthAndHeightPartial(Stream);
 #endif
 		}
 
+		partial void UpdatePixelWidthAndHeightPartial(Stream stream);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs:102]: For now this is no-op on Wasm :(

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**